contracts/Oracle.sol

```diff
@@ -144,6 +173,44 @@ contract Oracle is Ownable {
         oracleDisputePeriod[_oracle] = _disputePeriod;
 
         emit OracleDisputePeriodUpdated(_oracle, _disputePeriod);
-        (_oracle, _disputePeriod);
+    }
+
+    /**
+     * @notice set batch underlying asset price
+     * @dev underlying price can only be set after locking period is over and before starting dispute period
+     * @param _batch (hash of underlying, stike, collateral and expiry)
+     * @param _expiryTimestamp batch expiry timestamop
+     * @param _roundsBack number of chainlink price feed roundback
+     */
+    function setBatchUnderlyingPrice(
```

Comment:
when is this called?

Reply:
controller module.

Reply:
Why can't this be called by anyone?

Reply:
so that the Controller hash the batch and send expiry as arg based on the Otoken, in this way no dependency needed to Otoken in Oracle level + not random values+hashes from anyone.